### PR TITLE
Add roll angle to stylesheet

### PR DESCRIPTION
**Pull request by NathanMOlson**
Add Roll Angle

This PR adds roll angle to the stylesheet.

https://github.com/maplibre/maplibre-style-spec/issues/850
https://github.com/maplibre/maplibre-gl-js/issues/4717
https://github.com/maplibre/maplibre-gl-js/pull/4780

 - [X] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [X] Briefly describe the changes in this PR.
 - [X] Link to related issues.
 - [X] Write tests for all new functionality.
 - [X] Document any changes to public APIs.
 - [X] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

File: src/diff.test.ts [hunk 1]
```diff
@@ -302,6 +302,16 @@ describe('diff', () => {
         ]);
     });
 
+    test('set roll', () => {
```
Comment (HarelM) [inline src/diff.test.ts:305]: What about from roll 1 to undefined?

**Review by louwers — CHANGES_REQUESTED** [review]
Should indicate SDK support.

**Comment by HarelM** [comment]
Right. You can set it to 5.0 for js.

File: src/reference/v8.json [hunk 2]
```diff
@@ -54,6 +54,20 @@
...
+    "roll": {
+      "type": "number",
+      "default": 0,
+      "units": "degrees",
+      "doc": "Default roll, in degrees. The roll angle is measured counterclockwise about the camera boresight. The style roll will be used only if the map has not been positioned by other means (e.g. map options or user interaction).",
+      "example": 45,
+      "sdk-support": {
+        "basic functionality": {
+          "js": "5.0.0",
+          "android": "wontfix",
```
Comment (louwers) [inline src/reference/v8.json:66]: Why wontfix?

Comment (louwers) [reply src/reference/v8.json:66]: You can use https://github.com/maplibre/maplibre-native/issues/2941

Comment (NathanMOlson) [reply src/reference/v8.json:66]: I will update to link the newly created issue.

File: src/diff.test.ts (hunk 1 above)
Comment (NathanMOlson) [reply src/diff.test.ts:314]: Done.

**Comment by HarelM** [comment]
@louwers can I merge this?

**Review by louwers — APPROVED** [review]

**Pull request merged**